modules/cfe_testcase/src/fs_util_test.c

```diff
@@ -0,0 +1,161 @@
+                      CFE_FS_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(CFE_FS_ParseInputFileName(NameBuf, NULL, sizeof(NameBuf), CFE_FS_FileCategory_SCRIPT),
+                      CFE_FS_INVALID_PATH);
+    UtAssert_INT32_EQ(CFE_FS_ParseInputFileName(NameBuf, Name, 0, CFE_FS_FileCategory_SCRIPT), CFE_FS_BAD_ARGUMENT);
+
+    UtAssert_INT32_EQ(
+        CFE_FS_ParseInputFileNameEx(OutNameBuf, InNameBuf, sizeof(OutNameBuf), sizeof(InNameBuf), Name, Path, Ext),
+        CFE_SUCCESS);
+    UtAssert_StrCmp(ExpectedBuf, OutNameBuf, "Parse Input EX: %s", OutNameBuf);
+    UtAssert_INT32_EQ(
+        CFE_FS_ParseInputFileNameEx(OutNameBuf, NULL, sizeof(OutNameBuf), sizeof(InNameBuf), Name, Path, Ext),
+        CFE_SUCCESS);
+    UtAssert_StrCmp(ExpectedName, OutNameBuf, "Parse Input EX: %s", OutNameBuf);
+    UtAssert_INT32_EQ(CFE_FS_ParseInputFileNameEx(OutNameBuf, InNameBuf, sizeof(OutNameBuf), 0, Name, Path, Ext),
+                      CFE_SUCCESS);
+    UtAssert_StrCmp(ExpectedName, OutNameBuf, "Parse Input EX: %s", OutNameBuf);
+    UtAssert_INT32_EQ(
+        CFE_FS_ParseInputFileNameEx(OutNameBuf, InNameBuf, sizeof(OutNameBuf), sizeof(InNameBuf), NULL, Path, Ext),
+        CFE_SUCCESS);
+    UtAssert_StrCmp(ExpectedBuf, OutNameBuf, "Parse Input EX: %s", OutNameBuf);
+
+    UtAssert_INT32_EQ(
+        CFE_FS_ParseInputFileNameEx(NULL, InNameBuf, sizeof(OutNameBuf), sizeof(InNameBuf), Name, Path, Ext),
+        CFE_FS_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(CFE_FS_ParseInputFileNameEx(OutNameBuf, InNameBuf, 0, sizeof(InNameBuf), Name, Path, Ext),
+                      CFE_FS_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(
+        CFE_FS_ParseInputFileNameEx(OutNameBuf, NULL, sizeof(OutNameBuf), sizeof(InNameBuf), NULL, Path, Ext),
+        CFE_FS_INVALID_PATH);
+    UtAssert_INT32_EQ(CFE_FS_ParseInputFileNameEx(OutNameBuf, InNameBuf, sizeof(OutNameBuf), 0, NULL, Path, Ext),
+                      CFE_FS_INVALID_PATH);
+}
+
+void TestFileName(void)
+{
+    const char Path[] = "/func/FileName.test";
+    char       Name[OS_MAX_FILE_NAME];
+    const char ExpectedName[] = "FileName.test";
+
+    UtPrintf("Testing: CFE_FS_ExtractFilenameFromPath");
+
+    UtAssert_INT32_EQ(CFE_FS_ExtractFilenameFromPath(Path, Name), CFE_SUCCESS);
+    UtAssert_StrCmp(Name, ExpectedName, "Extract Filename: %s", Name);
+
+    UtAssert_INT32_EQ(CFE_FS_ExtractFilenameFromPath(NULL, Name), CFE_FS_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(CFE_FS_ExtractFilenameFromPath(Path, NULL), CFE_FS_BAD_ARGUMENT);
+}
+
+/* FT helper stub compatible with background file write DataGetter */
+bool FS_DataGetter(void *Meta, uint32 RecordNum, void **Buffer, size_t *BufSize)
+{
+    *Buffer  = NULL;
+    *BufSize = 0;
+    return true;
+}
+
+/* FT helper stub compatible with background file write OnEvent */
+void FS_OnEvent(void *Meta, CFE_FS_FileWriteEvent_t Event, int32 Status, uint32 RecordNum, size_t BlockSize,
+                size_t Position)
+{
+}
+
+CFE_FS_FileWriteMetaData_t State;
```

Comment:
wait a minute, I just noticed that "State" is both a global and a local stack variable.  Can we fix this?  

BTW global vars must be prefixed/qualified with an app name in some way to avoid collisions.  Collisions do happen and may be silent (no warning/error), you just see very strange results.  So if we go with a global var then please change the name too..

Reply:
Also - globals should be at the top of the file to comply with coding standards (this is why I didn't see it at first!!)